router/batchrouter/batchrouter.go

```diff
@@ -2397,16 +2397,23 @@ func (brt *HandleT) updateRudderSourcesStats(ctx context.Context, tx jobsdb.Upda
 }
 
 func (brt *HandleT) updateProcessedEventsMetrics(statusList []*jobsdb.JobStatusT) {
-	countPerState := map[string]int{}
+	eventsPerStateAndCode := map[string]map[string]int{}
 	for i := range statusList {
 		state := statusList[i].JobState
-		countPerState[state]++
-	}
-	for state, count := range countPerState {
-		stats.NewTaggedStat(`pipeline_processed_events`, stats.CountType, stats.Tags{
-			"module":   "batch_router",
-			"destType": brt.destType,
-			"state":    state,
-		}).Count(count)
+		code := statusList[i].ErrorCode
+		if _, ok := eventsPerStateAndCode[state]; !ok {
+			eventsPerStateAndCode[state] = map[string]int{}
+		}
+		eventsPerStateAndCode[state][code]++
+	}
+	for state, codes := range eventsPerStateAndCode {
+		for code, count := range codes {
+			stats.NewTaggedStat(`pipeline_processed_events`, stats.CountType, stats.Tags{
```

Comment:
` ==> "

Any particular reason for `?

Reply:
not really needed, but doesn't hurt either